### PR TITLE
Store timestamps in UTC

### PR DESCRIPTION
**Pull request by mauguignard**

### CODE REVIEW

File: sharedbuffers/mapped_struct.py [hunk 1]
```diff
@@ -2802,7 +2803,7 @@ def pack_into(cls, obj, buf, offs, idmap = None, implicit_offs = 0):
                 widmap.link(objid, obj)
 
         packer = cls.PACKER
-        timestamp = int(time.mktime(obj.timetuple()))
+        timestamp = int(calendar.timegm(obj.timetuple()))
```
Comment (klaussfreire) [inline sharedbuffers/mapped_struct.py]: This doesn't work right for datetime, but it does work for date.

I'd say revert datetime, and lets fix date for now.

File: sharedbuffers/mapped_struct.py [hunk 2]
```diff
@@ -2850,7 +2852,7 @@ def unpack_from(cls, buf, offs, idmap = None):
 
         packer = cls.PACKER
         timestamp, = packer.unpack_from(buf, offs)
-        rv =  date.fromtimestamp(timestamp)
+        rv =  date.fromtimestamp(time.mktime(time.gmtime(timestamp)))
```
Comment (klaussfreire) [inline sharedbuffers/mapped_struct.py]:
```suggestion
        rv =  datetime.utcfromtimestamp(timestamp).date()
```

That works reliably.

**Review by klaussfreire — CHANGES_REQUESTED** [review]

File: sharedbuffers/mapped_struct.py (hunk 2 above)
Comment (klaussfreire) [reply sharedbuffers/mapped_struct.py]: Try to add a few tests

File: sharedbuffers/mapped_struct.py (hunk 1 above)
Comment (mauguignard) [reply sharedbuffers/mapped_struct.py]: Done!

File: sharedbuffers/mapped_struct.py (hunk 2 above)
Comment (mauguignard) [reply sharedbuffers/mapped_struct.py]: Done!

File: sharedbuffers/mapped_struct.py [hunk 3]
```diff
@@ -2803,7 +2803,7 @@ def pack_into(cls, obj, buf, offs, idmap = None, implicit_offs = 0):
                 widmap.link(objid, obj)
 
         packer = cls.PACKER
-        timestamp = int(calendar.timegm(obj.timetuple()))
+        timestamp = int(time.mktime(obj.timetuple()))
```
Comment (klaussfreire) [inline sharedbuffers/mapped_struct.py:2806]: I think it needs to be timegm. There was a case that mktime didn't handle correctly.

Comment (mauguignard) [reply sharedbuffers/mapped_struct.py:2806]: This is for the `datetime` (that we talked about leaving it as it was). This commit was just a "rollback". `date` now uses `calendar.timegm`.

Comment (klaussfreire) [reply sharedbuffers/mapped_struct.py:2806]: Oh, sorry. Lost track of the context I guess.

**Review by klaussfreire — APPROVED** [review]

File: sharedbuffers/mapped_struct.py (hunk 3 above)
Comment (mauguignard) [reply sharedbuffers/mapped_struct.py:2806]: No worries! :raised_hands:

**Pull request merged**